File: appeal_tool/formatting.py
from __future__ import annotations

from appeal_tool.models import Parcel


def parcel_address(parcel: Parcel) -> str:
    city_zip = " ".join(part for part in (parcel.city.strip(), parcel.zip_code.strip()) if part)
    pieces = [part for part in (parcel.address.strip(), city_zip) if part]
    if not pieces:
        return "Not available from public dataset"
    return ", ".join(pieces)

File: src/domain/formatting.ts
import type { Parcel } from "./models";

export function parcelAddress(parcel: Parcel): string {
  const cityZip = [parcel.city.trim(), parcel.zipCode.trim()].filter(Boolean).join(" ");
  const pieces = [parcel.address.trim(), cityZip].filter(Boolean);
  if (pieces.length === 0) {
    return "Not available from public dataset";
  }
  return pieces.join(", ");
}
